### PR TITLE
[devops] add nix and npm publishing config

### PR DESCRIPTION
**Pull request by adrian-gierakowski**
noticed checksum missmatch when running ST build on new nix runner
without access to binary cache

published to npm (https://www.npmjs.com/package/@rhino.fi/starkware-crypto)
so that we can replace github reference with proper npm package

I've added auto-publish via github actions, but we'll sill need someone with admin access to add a github secrets to make it work

### CODE REVIEW

**Comment by arijoon** [comment]
The build is still failing on this, does it need to be actioned by someone with admin access to fix the build?

File: nix/pkgs.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            let
+              yarnExe = lib.getExe yarn-berry;
+            in utils.writeBashBin
+              name
+              ''
+              set -ueo pipefail
+
+              echo yarn version: $(${yarnExe} --version)
+
+              ${yarnExe} --immutable
```
Comment (arijoon) [inline nix/pkgs.nix:31]: is this expected to run in nix sandbox? since its not using nixify it'll fail without

**Review by arijoon — APPROVED** [review]

**Comment by adrian-gierakowski** [comment]
> The build is still failing on this, does it need to be actioned by someone with admin access to fix the build?

Yes, we need to add secret to gh actions

File: nix/pkgs.nix (hunk 1 above)
Comment (adrian-gierakowski) [reply nix/pkgs.nix:31]: No sandbox

**Pull request merged**